dfastmi/batch/ConfigurationCheckerLegacy.py

```diff
@@ -0,0 +1,426 @@
+        nwidth: float
+    ) -> Tuple[Vector, BoolVector, float, Vector, float, Vector, Vector, Vector]:
+        """
+        Determine discharges, times, etc. for version 1 analysis
+
+        Arguments
+        ---------
+        rivers : RiversObject
+            A dictionary containing the river data.
+        ibranch : int
+            Number of selected branch.
+        ireach : int
+            Number of selected reach.
+        config : configparser.ConfigParser
+            Configuration of the analysis to be run.
+        nwidth : float
+            Normal river width (from rivers configuration file) [m].
+
+        Return
+        ------
+        discharges : Vector
+            Array of discharges (Q); one for each forcing condition [m3/s].
+        apply_q : BoolVector
+            A list of flags indicating whether the corresponding entry in Q should be used.
+        q_threshold : float
+            River discharge at which the measure becomes active [m3/s].
+        time_mi : Vector
+            A vector of values each representing the fraction of the year during which the discharge Q results in morphological impact [-].
+        tstag : float
+            Fraction of year during which flow velocity is considered negligible [-].
+        fractions_of_the_year : Vector
+            A vector of values each representing the fraction of the year (T) during which the discharge is given by the corresponding entry in Q [-].
+        rsigma : Vector
+            A vector of values each representing the relaxation factor for the period given by the corresponding entry in Q [-].
+        celerity : Vector
+            A vector of values each representing the bed celerity for the period given by the corresponding entry in Q [m/s].
+        """
+        q_stagnant = reach.qstagnant
+        celerity_hg = reach.proprate_high
+        celerity_lw = reach.proprate_low
+
+        (
+            q_threshold,
+            Q1,
+            apply_q1,
+            tstag,
+            fractions_of_the_year,
+            rsigma,
+        ) = self._batch_get_discharges(
+            reach, config, q_stagnant, celerity_hg, celerity_lw, nwidth
+        )
+        discharges = Q1
+        apply_q = apply_q1
+        time_mi = tuple(0 if discharges[i] is None or discharges[i]<=q_stagnant else fractions_of_the_year[i] for i in range(len(fractions_of_the_year)))
+        celerity = (celerity_lw, celerity_hg, celerity_hg)
+
+        return (discharges, apply_q, q_threshold, time_mi, tstag, fractions_of_the_year, rsigma, celerity)
+
+    def _batch_get_discharges(
+        self,
+        reach : ReachLegacy,
+        config: configparser.ConfigParser,
+        q_stagnant: float,
+        celerity_hg: float,
+        celerity_lw: float,
+        nwidth: float,
+    ) -> Tuple[
+        Optional[float],
+        QRuns,
+        Tuple[bool, bool, bool],
+        float,
+        Vector,
+        Vector,
+    ]:
+        """
```

Comment:
this is a private method, do you want to keep the documentation for it, if yes, please update it.

Reply:
updated